apps/passport-server/src/services/issuanceService.ts

```diff
@@ -887,6 +893,25 @@ export class IssuanceService {
       return tickets;
     });
   }
+
+  /**
+   * Returns a promised verification of a PCD, either from the cache or,
+   * if there is no cache entry, from the multiprocess service.
+   */
+  private cachedVerify(serializedPCD: SerializedPCD): Promise<boolean> {
+    const key = JSON.stringify(serializedPCD);
+    const cached = this.verificationPromiseCache.get(key);
+    if (cached) {
+      return cached;
+    }
+    else {
+      const promise = this.multiprocessService.verifySignaturePCD(serializedPCD);
+      // If the promise rejects, delete it from the cache
+      promise.catch(() => this.verificationPromiseCache.delete(key));
+      this.verificationPromiseCache.set(key, promise);
```

Comment:
nit but there's a chance the promise rejects synchronously, so you should probably switch the order of these two lines